lib/backend/model/rule.go

```diff
@@ -46,3 +49,103 @@ type Rule struct {
 	NotICMPType    *int                  `json:"!icmp_type,omitempty" validate:"omitempty,gte=1,lte=255"`
 	NotICMPCode    *int                  `json:"!icmp_code,omitempty" validate:"omitempty,gte=1,lte=255"`
 }
+
+func (r Rule) String() string {
+	parts := make([]string, 0)
+	if r.Action != "" {
+		parts = append(parts, r.Action)
+	} else {
+		parts = append(parts, "allow")
+	}
+	if r.Protocol != nil {
+		parts = append(parts, r.Protocol.String())
+	}
+	if r.ICMPType != nil {
+		parts = append(parts, "type", strconv.Itoa(*r.ICMPType))
+	}
+	if r.ICMPCode != nil {
+		parts = append(parts, "code", strconv.Itoa(*r.ICMPCode))
+	}
+	if r.NotICMPType != nil {
+		parts = append(parts, "!type", strconv.Itoa(*r.NotICMPType))
+	}
+	if r.NotICMPCode != nil {
+		parts = append(parts, "!code", strconv.Itoa(*r.NotICMPCode))
+	}
+
+	if r.SrcTag != "" || len(r.SrcPorts) > 0 || r.SrcNet != nil || r.SrcSelector != "" ||
```

Comment:
Sounds good.

On Wed, 10 Aug 2016 21:46 Rob Brockbank, <notifications@github.com> wrote:

> In lib/backend/model/rule.go
> <https://github.com/tigera/libcalico-go/pull/55#discussion_r74327092>:
>
> > +		parts = append(parts, r.Protocol.String())
> > +	}
> > +	if r.ICMPType != nil {
> > +		parts = append(parts, "type", strconv.Itoa(*r.ICMPType))
> > +	}
> > +	if r.ICMPCode != nil {
> > +		parts = append(parts, "code", strconv.Itoa(*r.ICMPCode))
> > +	}
> > +	if r.NotICMPType != nil {
> > +		parts = append(parts, "!type", strconv.Itoa(*r.NotICMPType))
> > +	}
> > +	if r.NotICMPCode != nil {
> > +		parts = append(parts, "!code", strconv.Itoa(*r.NotICMPCode))
> > +	}
> > +
> > +	if r.SrcTag != "" || len(r.SrcPorts) > 0 || r.SrcNet != nil || r.SrcSelector != "" ||
>
> More maintainable if you keep separate 'from' and 'to' parts lists then at
> the end you can include the word "from" if there are any from parts, sim.
> for to. WDYT?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tigera/libcalico-go/pull/55/files/bf9f0db3ca1c7aa33f4feb11072c9750d8c0f4c7#r74327092>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAcpEKmPT5v39qGrJEjqjXVXJFndQ62Sks5qejihgaJpZM4Je6oQ>
> .
>